lm_eval/models/huggingface.py

```diff
@@ -337,6 +302,84 @@ def __init__(
                 f"Loglikelihood prefix token id used in evaluation: {self.prefix_token_id}"
             )
 
+    def _get_accelerate_args(
+        self,
+        parallelize: bool = None,
+        device_map: Optional[str] = "auto",
+        max_memory_per_gpu: Optional[Union[int, str]] = None,
+        max_cpu_memory: Optional[Union[int, str]] = None,
+        offload_folder: Optional[str] = "./offload",
+        gpus: Optional[int] = None,
+    ) -> dict:
+        """Returns the kwargs needed to apply `accelerate` in `AutoModel.from_pretrained`."""
+        num_local_processes = int(os.environ.get("LOCAL_WORLD_SIZE", 1))
+        num_machines = int(os.environ.get("WORLD_SIZE", 0)) // num_local_processes
+        if num_machines == 0:
+            eval_logger.info(
+                "We are not in a distributed setting. Setting model_parallel to False."
+            )
+            parallelize = False
+
+        if parallelize is None:
+            # If parallelism is unset by the user, we automatically assign model parallelism
+            # if enough extra GPUs are available
+            max_memory_all_gpus = get_max_memory()
+            # We just want gpu, not cpu, max memory
+            if "cpu" in max_memory_all_gpus:
+                del max_memory_all_gpus["cpu"]
+            model_parallel = bool(num_local_processes < len(max_memory_all_gpus))
```

Comment:
```suggestion
            parallelize = bool(num_local_processes < len(max_memory_all_gpus))
```
this needs to propagate back to `parallelize`, no?